cpp/dolfinx/fem/utils.h

```diff
@@ -779,6 +756,87 @@ Form<T, U> create_form_factory(
     }
   }
 
+  // Attach vertex kernels
+  {
+    for (std::size_t form_idx = 0; form_idx < ufcx_forms.size(); ++form_idx)
+    {
+      const ufcx_form& form = ufcx_forms[form_idx];
+
+      std::span<const int> ids(form.form_integral_ids
+                                   + integral_offsets[vertex],
+                               num_integrals_type[vertex]);
+      auto sd = subdomains.find(IntegralType::vertex);
+      for (int i = 0; i < num_integrals_type[vertex]; ++i)
+      {
+        const int id = ids[i];
+        ufcx_integral* integral
+            = form.form_integrals[integral_offsets[vertex] + i];
+        assert(integral);
+        check_geometry_hash(*integral, form_idx);
+
+        std::vector<int> active_coeffs;
+        for (int j = 0; j < form.num_coefficients; ++j)
+        {
+          if (integral->enabled_coefficients[j])
+            active_coeffs.push_back(j);
+        }
+
+        impl::kernel_t<T, U> k = impl::extract_kernel<T>(integral);
+        assert(k);
+
+        // Build list of entities to assembler over
+        auto v_to_c = topology->connectivity(0, tdim);
+        assert(v_to_c);
+        auto c_to_v = topology->connectivity(tdim, 0);
+        assert(c_to_v);
+
+        // pack for a range of vertices a flattened list of cell index c_i and
+        // local vertex index l_i:
+        //  [c_0, l_0, ..., c_n, l_n]
+        auto get_cells_and_vertices = [v_to_c, c_to_v](auto vertices_range)
+        {
+          std::vector<std::int32_t> cell_and_vertex;
+          cell_and_vertex.reserve(2 * vertices_range.size());
+          for (std::int32_t vertex : vertices_range)
+          {
+            std::array<std::int32_t, 2> pair = impl::get_cell_vertex_pairs<1>(
```

Comment:
Is there a reason for `get_cell_vertex_pairs` not having the same kind of call signature as `get_cell_facet_pairs`, i.e. that `v_to_c` is passed into the function, and there `v_to_c->links(vertex)` is extracted?